File: CreationalPatterns/Builder/TypeScript/CarBuilder.ts
import { Builder } from './Builder';
import { Car } from './Car';

export class CarBuilder implements Builder {
    private car: Car = new Car();

    reset(): void {
        this.car = new Car();
    }
    setSeats(seats: number): void {
        this.car.seats = seats;
    }
    setEngine(engine: string): void {
        this.car.engine = engine;
    }
    setTripComputer(hasTripComputer: boolean): void {
        this.car.tripComputer = hasTripComputer;
    }
    setGPS(hasGPS: boolean): void {
        this.car.gps = hasGPS;
    }
    getResult(): Car {
        return this.car;
    }
}

File: CreationalPatterns/Builder/Go/car_builder.go
package main

type CarBuilder struct {
	car *Car
}

func (b *CarBuilder) Reset() {
	b.car = &Car{}
}

func (b *CarBuilder) SetSeats(seats int) {
	b.car.Seats = seats
}

func (b *CarBuilder) SetEngine(engine string) {
	b.car.Engine = engine
}

func (b *CarBuilder) SetTripComputer(hasTripComputer bool) {
	b.car.TripComputer = hasTripComputer
}

func (b *CarBuilder) SetGPS(hasGPS bool) {
	b.car.GPS = hasGPS
}

func (b *CarBuilder) GetResult() *Car {
	return b.car
}
